### PR TITLE
async call gas used fixes

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: arwen/contexts/metering.go [hunk 1]
```diff
@@ -236,6 +236,13 @@ func (context *meteringContext) GasUsedByContract() (uint64, uint64) {
 	return gasUsed, remainedFromForwarded
 }
 
+// GasUsedForExecution returns the actual gas used for execution for the contract which needs to be restored
+func (context *meteringContext) GasUsedForExecution() uint64 {
+	gasUsed, _ := context.GasUsedByContract()
+	gasUsed, _ = math.SubUint64(gasUsed, context.initialCost)
```
Comment (SebastianMarian) [inline arwen/contexts/metering.go:242]: Could overflow this subtraction in some edge case situation? SafeSub ?

File: arwen/common.go [hunk 2]
```diff
@@ -33,9 +33,12 @@ const (
 	// with its corresponding callback
 	SyncCall AsyncCallExecutionMode = iota
 
-	// AsyncBuiltinFunc indicates that the async call is a cross-shard call to a
+	// AsyncBuiltinFuncCrossShard indicates that the async call is an intra-shard built in function call
```
Comment (andrei-marinica) [inline arwen/common.go:36]: typo in comment, should be AsyncBuiltinFuncIntraShard

Comment (BeniaminDrasovean) [inline arwen/common.go:36]: Update comment.

**Review by BeniaminDrasovean — DISMISSED** [review]

File: arwen/common.go (hunk 2 above)
Comment (sasurobert) [reply arwen/common.go:36]: comment is correct

Comment (BeniaminDrasovean) [reply arwen/common.go:36]: The comment is with crossShard, but the constant with intaShard

File: arwen/contexts/metering.go (hunk 1 above)
Comment (sasurobert) [reply arwen/contexts/metering.go:242]: this means safeSub

**Review by ccorcoveanu — APPROVED** [review]

File: arwen/contexts/metering.go (hunk 1 above)
Comment (SebastianMarian) [reply arwen/contexts/metering.go:242]: In method computeGasUsedInExecutionBeforeReset it is used core.SafeSubUint64 this is why I asked

**Review by SebastianMarian — DISMISSED** [review]

**Review by andrei-marinica — DISMISSED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by ccorcoveanu — APPROVED** [review]

**Pull request merged**